engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -76,4 +96,11 @@ public void resetCamera(ResetCameraEvent resetCameraEvent, EntityRef client, Aut
         }
         Location.attachChild(targetEntityForCamera, clientComponent.camera, cameraPosition, new Quat4f(Quat4f.IDENTITY));
```

Comment:
Regarding the 1 vs 2 attachChild calls: my thinking is that less code to read (in such a simple case) is less code to understand, hence my suggestion. But I'm not sure what the purists of code would suggest in this case. Another way to tackle this, if there was a condition more likely than the other, would be to initialize the cameraPosition with the typical value and then handle the more atypical case in an IF block without ELSE, leaving a single attachChild call. 

But you mention there is no guarantee that the camera entity will come with a LocationComponent. Why is that the case? I mean, what good would a camera be without LocationComponent? Or are you referring to cameras being in-transition between one entity and another?